src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -1075,6 +1075,21 @@ private static SupportedPlatform GetCurrentRunningOS()
                 ThrowExitApplicationException(ExitReason.NoRulesLoaded);
             }
 
+            ISet<string> disabledSkimmers = BuildDisabledSkimmersSet(context, skimmers);
+
+            if (disabledSkimmers.Count == skimmers.Count())
+            {
+                Errors.LogAllRulesExplicitlyDisabled(context);
+                ThrowExitApplicationException(ExitReason.NoRulesLoaded);
+            }
+
+            this.CheckIncompatibleRules(skimmers, context, disabledSkimmers);
+
+            MultithreadedAnalyzeTargets(context, skimmers, disabledSkimmers);
```

Comment:
>MultithreadedAnalyzeTargets [](http://example.com/codeflow?start=12&length=27)

the real question, since you are in this code is why do we pass the skimmers as an argument when the context object contains a skimmer property?

the second question I understand, why aren't the disabled skimmers a property on the context? well, they could be. but the disabled skimmers by definition are a cross-thread thing and we are nervous about cross-thread data in the context object.

but we could simply do this. add the disabled skimmers onto the context object and pass it everywhere.

you've still built/exposed a nice helper.